serverlessworkflow/modules/ROOT/pages/eventing/working-with-callbacks.adoc

```diff
@@ -1,26 +1,43 @@
+For the Serverless Workflow to identify the published CloudEvent it is waiting for, the external service developer includes the process instance ID in the CloudEvent header or use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. 
 
-From the flow perspective, async means that the control will be returned to the caller almost immediately, without waiting for the actual operation to be finished. Once it ends, a link:{cloud_events_url}[Cloud Event] will be published to resume the flow.
+The Serverless Workflow correlation mechanism document focuses on the former mechanism that is based on the fact that each Serverless Workflow instance contains a unique identifier, which is generated automatically.
 
-For the flow to know this published event is the one it is waiting for, the external service developer might include the Kogito process instance id in the event header or, preferably, use the xref:event-correlation-with-workflows[Serverless Workflow correlation mechanism]. This guide will focus on the former mechanism, which is based on the fact that every Kogito Workflow instance has an automatically generated and unique identifier.
+[[ref-example-callback]]
+== Example of Callback state
 
-== Error callback example 
+To understand the Callback state, see the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[`serverless-workflow-callback-quarkus`] example application in GitHub repository. 
 
-The workflow in the link:{kogito_sw_examples_url}/serverless-workflow-callback-quarkus[Serverless Workflow Callback Quarkus] example illustrates the callback feature. Once started (this workflow does not have an initial model), it publishes a Cloud Event of type `resume` and waits for a Cloud Event of `wait` type.
+The Serverless Workflow in the `serverless-workflow-callback-quarkus` example application does not consist of an initial model. However, once the workflow is started, it publishes a CloudEvent of the `resume` type and waits for a CloudEvent, which contains the type `wait`.
 
-The Cloud Event with type `resume` is consumed by a listener that simulates the behavior of an external service. Consequently, as an external service will do, once the actions associated with that event reception have finished, this listener publishes a new Cloud Event with type `wait`. Once this event is received, the workflow moves to the next state and successfully ends. 
+A listener consumes the CloudEvent with the `resume` type, which affects the behavior of an external service. Consequently, on the external service side, when the actions associated with the `resume` type CloudEvent are completed, the listener publishes a `wait` type CloudEvent. Once the `wait` type CloudEvent is received, the workflow moves to the next state and ends successfully.
```

Comment:
Again, completely change the meaning. 
We are defining a listener that simulates the behaviour of an external service. Now it seems that there is a litener that affects the behaviour of an extenal service (which does not really exist in this example)

Reply:
Done, changed the sentence as follows:

A listener consumes the CloudEvent with the `resume` type, which simulates the behavior of an external service.

Reply:
PRoblem is that what this "which" refers to. It is the listener that simulated the behaviour of an external service.

Reply:
If there is confusion with `which`. We can change the sentence to:

A listener consumes the CloudEvent with the resume type and simulates the behavior of an external service.